### ISSUE TITLE
models usage 

### ISSUE THREAD

**Issue by HeChengHui** [hnuzhy/DirectMHP#28]
Thank you for providing the codes and pre-trained models. 

I am looking through the demos folder and saw that `SixDRepNet_AGORA_bs256_e100_epoch_last.pth` and `SixDRepNet_CMU_bs256_e100_epoch_last.pth` can be used in _image_vis3d_6DRepNet.py_. However, when I looked into _image_vis3d.py_ and _video_vis3d.py_, it seems like it is using [yolov5m](https://github.com/hnuzhy/DirectMHP/blob/737da6af9f31b44fa69023605667de726032c237/demos/video_vis3d.py#L188). In that case, where am I to use the models stated under [Single HPE Task Pretrained on WiderFace and Finetuning on 300W-LP](https://huggingface.co/HoyerChou/DirectMHP#single-hpe-task-pretrained-on-widerface-and-finetuning-on-300w-lp)?

**Comment by hnuzhy** [comment]
For inference, please refer https://github.com/hnuzhy/DirectMHP#inference
You can change the `--weights` file by giving it a new argument.

**Comment by HeChengHui** [comment]
@hnuzhy 
Thank you for the clarification. The results are very good, even with masked faces.
In your code, it seems like face detection is also part of the model inference, as it outputs [face bbxs](https://github.com/hnuzhy/DirectMHP/blob/737da6af9f31b44fa69023605667de726032c237/demos/video.py#L95) as well. Is it possible for me to use my own face detection model and feed the cropped face image to the model instead? would that be a sensible approach?

**Comment by hnuzhy** [comment]
You are right. DirectMHP can do head detection and its pose estimation at the same time. Therefore, for an undetected face/head, we may also lose its head pose. For these cases, you can directly use single HPE method like 6DRepNet, which will always give you an estimated head pose for the input.

**Comment by HeChengHui** [comment]
@hnuzhy 
I see. Shame that it cannot be decoupled.

I have tried 6DRepNet from [here](https://github.com/hnuzhy/DirectMHP/blob/main/exps/sixdrepnet.zip) with the models from [here](https://huggingface.co/HoyerChou/DirectMHP#single-hpe-sixdrepnet-re-trained-on-agora-hpe-and-cmu-hpe), but the results are very bad; even worse then the original 6DRepNet. Is there a difference between the mentioned code and the one found [here](https://github.com/hnuzhy/DirectMHP/blob/main/demos/image_vis3d_6DRepNet.py)?

**Comment by hnuzhy** [comment]
> @hnuzhy I see. Shame that it cannot be decoupled.
> 
> I have tried 6DRepNet from [here](https://github.com/hnuzhy/DirectMHP/blob/main/exps/sixdrepnet.zip) with the models from [here](https://huggingface.co/HoyerChou/DirectMHP#single-hpe-sixdrepnet-re-trained-on-agora-hpe-and-cmu-hpe), but the results are very bad; even worse then the original 6DRepNet. Is there a difference between the mentioned code and the one found [here](https://github.com/hnuzhy/DirectMHP/blob/main/demos/image_vis3d_6DRepNet.py)?

I have retrained 6DRepNet for the full-range HPE task. But results are not that good. Maybe it need more refined adjustments of parameters.
